### PR TITLE
feat(notch): collapsed-notch deep links and session pairing

### PR DESCRIPTION
**Pull request by sk-ruban**
## Summary

The collapsed notch now behaves as a coherent unit — what it displays, what clicking it does, and what the expanded panel opens to all agree.

- **Ring click → usage detail.** Clicking the usage ring expands the panel directly to the usage view, on the ring's own provider. Plain notch clicks always open the activity view (the panel no longer remembers the usage view across collapse).
- **Mascot click → its session.** With multiple sessions, clicking the mascot selects that session and opens its activity directly instead of the session picker.
- **Ring pairs with the mascot.** The ring shows usage for the session the sprite slot actually displays (previously it followed the sticky selected session, so a Codex ring could sit next to a Claude mascot). When idle, both fall back to the last-used agent provider.
- **Last-viewed session pins the collapsed notch.** After viewing a session in the panel and collapsing, the notch keeps showing it until another session produces a newer event.
- **Usage detail defaults to last-used provider** when idle, with the dataless-provider reroute consolidated in `UsageDetailView.resolvedProvider`; both usage services now expose a shared `hasUsageData`.

## Fixes found in review

- Ring tap target didn't track the ring visually (`contentShape`/gesture sat outside the `offset` transform).
- The deep link depended on the mouse monitor's deferred-Task timing; the gesture now requests expansion itself.

## Testing

- 17 new/updated unit tests across `ExpandedPanelViewTests`, `NotchContentViewTests`, `UsageDetailViewTests`, `SessionStoreTests` (provider defaults, reroute rule, ring pairing, selection pin incl. boundary cases).
- `./scripts/test.sh focused` and the touched suites all pass; manually verified ring/mascot click-through, provider pairing, and the pin flow.

### CODE REVIEW

File: notchi/Tests/UsageDetailViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    func testResolvedProviderReroutesDatalessCodexToClaudeWithData() {
+        XCTAssertEqual(
+            UsageDetailView.resolvedProvider(selected: .codex, claudeHasData: false, codexHasData: true),
+            .codex
+        )
+        XCTAssertEqual(
+            UsageDetailView.resolvedProvider(selected: .codex, claudeHasData: true, codexHasData: false),
+            .claude
+        )
+    }
```
[the remark this answers is not in the document]
Comment (sk-ruban) [reply notchi/Tests/UsageDetailViewTests.swift:33]: Fixed in edb0ded — the assertion covered a distinct no-reroute row (selected Codex has data, Claude doesn't), so it moved into `testResolvedProviderKeepsSelectionWhenItHasData` rather than being deleted.

**Comment by sk-ruban** [comment]
Re: **Gesture hit area may not follow the visually offset ring** (P1, `NotchContentView.swift` ringSlot)

The hit area does follow the ring in this order. `.offset` doesn't move the layout frame, but hit-testing recurses through the offset transform for modifiers applied inside it — the mismatch only occurs when `contentShape` is applied *outside* `.offset`, which is the order this PR removed. Verified empirically with synthetic `sendEvent` clicks: `contentShape → gesture → scaleEffect → offset` fires at the visual position and not at the pre-offset position.
